chain-time/src/timeframe.rs

```diff
@@ -0,0 +1,168 @@
+use crate::timeline::Timeline;
+use std::time::{Duration, SystemTime};
+
+/// Identify a slot in a *specific* timeframe
+///
+/// The slots are not comparable to others slots made on a
+/// different time frame
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Slot(pub(crate) u64);
+
+impl From<Slot> for u64 {
+    fn from(s: Slot) -> u64 {
+        s.0
+    }
+}
+
+/// Identify a slot in a specific timeframe and a leftover duration
+#[derive(Debug)]
+pub struct SlotAndDuration {
+    pub slot: Slot,
+    /// The offset of a specific time frame in
+    pub slot_duration_offset: Duration,
+}
+
+/// Time frame which is a timeline that is configured to be split in discrete slots
+pub struct TimeFrame {
+    timeline: Timeline,
+    slot_offset: Slot,
+    slot_duration: NbSeconds,
+    //slot_configuration: Vec<u64>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NbSeconds(u64);
+
+impl NbSeconds {
+    pub fn new(seconds: u32) -> Self {
```

Comment:
Continuing with the above, naming the constructor `SlotDuration::from_secs` would clarify the input units and match with `Duration::from_secs`. The internal representation is not public.